overlord/devicestate/handlers.go

```diff
@@ -63,13 +68,54 @@ func deviceAPIBaseURL() string {
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength        uint64 = 4096
+	retryInterval           = 60 * time.Second
+	deviceAPIBase           = deviceAPIBaseURL()
+	requestIDURL            = deviceAPIBase + "request-id"
+	serialRequestURL        = deviceAPIBase + "devices"
+	sshKeyFile              = "/tmp/snapd.key.tmp"
+	sshPublicKeyFile        = sshKeyFile + ".pub"
 )
 
+func sshGenerateKey() (*rsa.PrivateKey, error) {
+	defer func() {
+		os.Remove(sshKeyFile)
+		os.Remove(sshPublicKeyFile)
+	}()
+
+	os.Remove(sshKeyFile)
+	os.Remove(sshPublicKeyFile)
+
+	cmd := exec.Command("ssh-keygen", "-t", "rsa", "-b", strconv.FormatUint(keyLength, 10), "-N", "", "-f", sshKeyFile)
+
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		return nil, err
```

Comment:
we probably want to use osutil.OutputErr here:

out, err := cmd.CombinedOutput()
if err != nil {
   return nil, osutil.OutputErr(out, err)
}